### PR TITLE
Mass code improvements

### PR DESCRIPTION
**Pull request by SimHoZebs**
Change details can be found in the commit message.
Few of the changes addresses discussion #6.
Feel free to ask questions about the changes.

### CODE REVIEW

File: pages/api/socket.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+type Res = NextApiResponse & {
+  socket: NetSocket & {
+    server: HTTPServer & {
+      io?: IOServer;
+    };
+  };
+};
+
+export default function handler(
```
Comment (ipod1g) [inline pages/api/socket.ts:15]: Are we handling all the backend game logic in this handler?

File: lib/components/ServerTest.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import React, { useEffect, useState } from "react";
+import { io, Socket } from "socket.io-client";
+
+let socket: Socket;
+const ServerTest = () => {
```
Comment (ipod1g) [inline lib/components/ServerTest.tsx:5]: So this file is a basic boilerplate for me to fiddle with and experiment right?

Comment (SimHoZebs) [reply lib/components/ServerTest.tsx:5]: This is an example component I made that showcases how the front end can interact with the backend code. You can use it as a reference to build components that can interact with it. My bad on the lack of function description.

File: pages/api/socket.ts (hunk 1 above)
Comment (SimHoZebs) [reply pages/api/socket.ts:15]: `handler`s in API routes handle everything server request. Here's a good read: https://nextjs.org/docs/api-routes/introduction

As for this route, you are correct. This is where all the socket interaction will happen.

File: pages/index.tsx [hunk 3]
```diff
@@ -1,15 +1,132 @@
...
+import React, { useState, useRef } from "react";
+import Board from "../lib/components/Board";
+import boardWithRandomlyPlacedShips from "../lib/functions/boardWithRandomlyPlacedShips";
+import { shipLength } from "../lib/interfaces/types";
+import ServerTest from "../lib/components/ServerTest";
+import Ship from "../lib/components/Ship";
+
+//Conditions 'Sunk' | 'Hit' | 'Miss' | 'Ship' | 'Empty'
 
 const Home: NextPage = () => {
```
[the remark this answers is not in the document]
Comment (SimHoZebs) [reply pages/index.tsx:11]: It replaces stages, correct. However, it has nothing to do with a page redirect; it is the index page. 

I think we should discuss more on how you want to manage stages. This is what I implemented after our last discussion, as see no need to have multiple components that represent the same game. We just need to swap out small parts of the logic instead.

**Pull request merged**